src/Normalizer/StrawberryfieldFieldItemNormalizer.php

```diff
@@ -67,19 +68,42 @@ public function denormalize($data, $class, $format = NULL, array $context = [])
 
     /** @var \Drupal\Core\Field\FieldItemInterface $field_item */
     $field_item = $context['target_instance'];
-    $this->checkForSerializedStrings($data, $class, $field_item);
 
-    $field_item->setValue($this->constructValue($data, $context));
+    $this->checkForSerializedStrings($data, $class, $field_item);
+    
+    // Set each key in the field_item with its constructed value
+    // At this point all values will be encoded JSON strings
+    // Coming from minio data, keys should be 'value' and 'str_flatten_keys'
+    $constructedValue = $this->constructValue($data, $context);
+    foreach ($constructedValue as $key => $value) {
+      $field_item->set($key, $value);
+    }
+    
     return $field_item;
   }
 
-
-/**
-   * {@inheritdoc}
-   */
+  /**
+   * Encodes the $data items that are not already strings.
+   * 
+   * @param mixed $data
+   * @param array $context
+   * 
+   * @return array|mixed
+   */  
   protected function constructValue($data, $context) {
-    //$data = $this->serializer->encode($data, 'json');
-    return parent::constructValue($data, $context);
+    // Encode individually, otherwise 'value' gets nested and breaks the SBF
+    $individualEncodedValues = [];
+    foreach ($data as $key => $value ) {
+      $isJSONString = is_string($value) && is_array(json_decode($value, true)) && (json_last_error() == JSON_ERROR_NONE);
```

Comment:
Can we check here if a property is calculated + readonly ? And based on that avoid setting it?